nni/contrib/compression/utils/docstring.py

```diff
@@ -0,0 +1,66 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+
+_EVALUATOR_DOCSTRING = r"""NNI will use the evaluator to intervene in the model training process,
+        so as to perform training-aware model compression.
+        All training-aware model compression will use the evaluator as the entry for intervention training in the future.
+        Usually you just need to wrap some classes with ``nni.trace`` or package the training process as a function to initialize the evaluator.
+        Please refer :doc:`/compression/compression_evaluator` for a full tutorial on how to initialize a ``evaluator``.
+
+        The following are two simple examples, if you use native pytorch, please refer to :class:`nni.contrib.compression.TorchEvaluator`,
+        if you use pytorch_lightning, please refer to :class:`nni.contrib.compression.LightningEvaluator`,
+        if you use huggingface transformer trainer, please refer to :class:`nni.contrib.compression.TransformersEvaluator`::
+
+            # LightningEvaluator example
+            import pytorch_lightning
+            lightning_trainer = nni.trace(pytorch_lightning.Trainer)(max_epochs=1, max_steps=50, logger=TensorBoardLogger(...))
+            lightning_data_module = nni.trace(pytorch_lightning.LightningDataModule)(...)
+
+            from nni.contrib.compression import LightningEvaluator
+            evaluator = LightningEvaluator(lightning_trainer, lightning_data_module)
+
+            # TorchEvaluator example
+            import torch
+            import torch.nn.functional as F
+
+            def training_model(model, optimizer, training_step, lr_scheduler, max_steps, max_epochs, *args, **kwargs):
```

Comment:
it would be better to explicitly say that user customized `training_model` should have follow this signature, and `training_step` should be defined out of `training_model`.

Reply:
good suggestion, fixed